python_modules/dagster/dagster/_core/definitions/asset_reconciliation_sensor.py

```diff
@@ -741,8 +739,11 @@ def conditions_for_candidate(
             conditions.add(ParentMaterializedAutoMaterializeCondition())
 
         # if the parents will not be resolved this tick
-        if get_unresolved_parent_asset_keys(candidate):
-            conditions.add(ParentOutdatedAutoMaterializeCondition())
+        waiting_on_asset_keys = get_waiting_on_asset_keys(candidate)
+        if waiting_on_asset_keys:
+            conditions.add(
+                ParentOutdatedAutoMaterializeCondition(waiting_on_asset_keys=waiting_on_asset_keys)
```

Comment:
If I understand correctly, this will include all the ancestors that we're waiting on, not just the ones at the root of the ancestor graph?

Is that what we want to show on the frontend?

I generally think the frontend here should be pretty dumb, and if we're going to filter the intermediate assets out, we should do it here, not there.

Reply:
ah so the function on the caching instance queryer actually does that filtering already -- so if you have:

`A -> B -> C -> D -> E`, and `A` has just been updated, the set of keys that we're waiting on would be identical for `C`, `D`, or `E` (always just `B`). I added a test confirming this behavior.

Updated the name of the function to be a bit clearer here